compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -615,6 +615,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             return false;
         }
         let pin_did = self.tcx.lang_items().pin_type();
+        // FIXME: replace mk_box with?
```

Comment:
`Ty::new_box` as per below?

Reply:
There still no `unwrap` here, should it be `This guards the new_box below.` or remove comment completely?

Reply:
added review commit.